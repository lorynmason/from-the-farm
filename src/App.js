import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { Buy } from './Components/Buy';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Buy />
      </div>
    );
  }
}

export default App;
