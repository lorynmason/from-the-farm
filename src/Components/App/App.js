import React, { Component } from 'react';
import { Buy } from '../Buy/Buy';
import { Header } from '../Header/Header';
import '../../styles/main.scss';
import { Switch, Route, withRouter } from 'react-router';
import Profile from '../Profile/Profile';
import { createBrowserHistory } from 'history';
import { connect } from 'react-redux';
import { addUser, removeUser, addProducts, addMessage } from '../../actions';
import { fetchVendors } from '../../thunks/fetchVendors';
import Message from '../Message/Message';

class App extends Component {

  async componentDidMount() {
    this.props.fetchVendors('https://xpoll-be.herokuapp.com/api/v1/vendors');
    this.props.addUserToStore({
      address: "85 Hooker St",
      bio: "Orange You Glad You Didn't Say Bananas",
      city: "denver",
      email: "ruffnbuff@example.com",
      id: 6,
      lat: 39.717646,
      long: -105.029438,
      name: "Oranges 4 Eva",
      phone: 2313414141,
      state: "CO"
    })
  }

  search = async({ productId, location, range }) => {
    const baseUrl = 'https://xpoll-be.herokuapp.com/api/v1/search'
    let path;
    if (!location) {
      console.log('error: please enter a search location')
      this.props.addMessage('error: please enter a search location')
    } 
    
    if (!range) {
      range = "50"
    } 
    
    if (!productId) {
      path = `?loc=${location}&range=${range}`
    } else {
      path = `?loc=${location}&range=${range}&item=${productId}`
    }
  
    const response = await this.props.fetchVendors(baseUrl + path)
    this.filterProducts(productId, response)
  }

  filterProducts = (id, response) => {
    if (id) {
      const newProducts = this.props.products.filter( product => {
        id = parseInt(id)
        return product.item_id === id
      });
      
      this.props.addProductsToStore(newProducts);
    }
  }

  render() {
    const history = createBrowserHistory();
    return (
      <div className="App">
        <Header />
        <Message />
        <Switch>
          <Route path="/buy" render={() => <Buy history={history} search={this.search}/>}/>
          <Route path="/profile" render={() => <Profile user={this.props.user} products={this.props.products} />}/>
        </Switch>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  vendors: state.vendors,
  products: state.products,
  user: state.user
});

export const mapDispatchToProps = (dispatch) => {
  return {
    addUserToStore: (user) => dispatch(addUser(user)),
    fetchVendors: (url) => dispatch(fetchVendors(url)),
    addProductsToStore: (products) => dispatch(addProducts(products)),
    addMessage: (message) => dispatch(addMessage(message))
  } 
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
