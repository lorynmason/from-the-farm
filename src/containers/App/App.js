import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Switch, Route, withRouter, Redirect } from 'react-router';
import Buy from '../Buy/Buy';
import AddProductForm from '../AddProductForm/AddProductForm';
import Header from '../Header/Header';
import '../../styles/main.scss';
import Profile from '../Profile/Profile';
import { fetchVendors } from '../../thunks/fetchVendors';
import Message from '../Message/Message';
import { About } from '../About/About';
import Login from '../../containers/Login/Login';

export class App extends Component {
  async componentDidMount() {
    this.props.fetchVendors('https://xpoll-be.herokuapp.com/api/v1/vendors');
  }

  render() {
    return (
      <div className="App">
        <Header />
        <Message />
        <Switch>
          <Route exact path="/" component={Buy} />
          <Route path="/about" component={About}/>
          <Route path="/login" component={Login}/>
          <Route path="/buy" component={Buy} />
          <Route path="/add-product" render={() => this.props.user.name ? (<AddProductForm />) : (<Redirect to="/login" />)} />
          <Route path="/profile" render={() => this.props.user.name ? (<Profile />) : (<Redirect to="/login" />)} />
        </Switch>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  vendors: state.vendors,
  products: state.products,
  user: state.user,
  isLoading: state.isLoading
});

export const mapDispatchToProps = (dispatch) => {
  return {
    fetchVendors: url => dispatch(fetchVendors(url))
  };
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
