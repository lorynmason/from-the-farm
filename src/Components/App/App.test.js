import React from 'react';
import App from './App';
import { shallow } from 'enzyme'

describe('App', () => {
  it('should match snapshot', () => {
    let wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  });
});
