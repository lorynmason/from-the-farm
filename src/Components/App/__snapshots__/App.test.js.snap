// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <Buy
          coordinates={
            Array [
              Array [
                39.7392,
                -104.9903,
              ],
              Array [
                40.015,
                -105.2705,
              ],
              Array [
                39.6133,
                -105.0166,
              ],
              Array [
                39.7294,
                -104.8319,
              ],
            ]
          }
        />,
        <ProductList />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "coordinates": Array [
            Array [
              39.7392,
              -104.9903,
            ],
            Array [
              40.015,
              -105.2705,
            ],
            Array [
              39.6133,
              -105.0166,
            ],
            Array [
              39.7294,
              -104.8319,
            ],
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <Buy
            coordinates={
              Array [
                Array [
                  39.7392,
                  -104.9903,
                ],
                Array [
                  40.015,
                  -105.2705,
                ],
                Array [
                  39.6133,
                  -105.0166,
                ],
                Array [
                  39.7294,
                  -104.8319,
                ],
              ]
            }
          />,
          <ProductList />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "coordinates": Array [
              Array [
                39.7392,
                -104.9903,
              ],
              Array [
                40.015,
                -105.2705,
              ],
              Array [
                39.6133,
                -105.0166,
              ],
              Array [
                39.7294,
                -104.8319,
              ],
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
